embedded-graphics/src/primitives/arc.rs

```diff
@@ -0,0 +1,632 @@
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Arc, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledArcIterator<C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledArcIterator::new(self)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSector {
+    line_a: LinearEquation,
+    line_b: LinearEquation,
+    draw_above_a: bool,
+    draw_above_b: bool,
+    sweep: Angle,
+}
+
+impl PlaneSector {
+    fn new(arc: &Arc) -> Self {
+        let angle_start = arc.angle_start.normalize_from(-ANGLE_90DEG);
+        let angle_end = arc.angle_end().normalize_from(-ANGLE_90DEG);
+        let negative_sweep = arc.angle_sweep < Angle::zero();
+
+        Self {
+            line_a: LinearEquation::from_point_angle(arc.center(), arc.angle_start),
+            line_b: LinearEquation::from_point_angle(arc.center(), arc.angle_end()),
+            draw_above_a: (angle_start < ANGLE_90DEG) ^ negative_sweep,
+            draw_above_b: (angle_end >= ANGLE_90DEG) ^ negative_sweep,
+            sweep: arc.angle_sweep.abs(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            line_a: LinearEquation::flat(),
+            line_b: LinearEquation::flat(),
+            draw_above_a: true,
+            draw_above_b: true,
+            sweep: Angle::zero(),
+        }
+    }
+}
+
+/// Iterator that returns only the points which are inside a plane sector defined by two lines.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSectorIterator {
+    plane_sector: PlaneSector,
+    points: super::rectangle::Points,
+}
+
+impl PlaneSectorIterator {
+    fn new(arc: &Arc) -> Self {
+        Self {
+            plane_sector: PlaneSector::new(arc),
+            points: arc.bounding_box().points(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            plane_sector: PlaneSector::empty(),
+            points: Rectangle::new(Point::zero(), Size::zero()).points(),
+        }
+    }
+}
+
+impl Iterator for PlaneSectorIterator {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let plane_sector = self.plane_sector;
+        self.points.find(|p| {
+            let side_a = plane_sector.line_a.side(p);
+            let side_b = plane_sector.line_b.side(p);
+
+            let correct_a_side = plane_sector.draw_above_a ^ (side_a == LineSide::BELLOW);
+            let correct_b_side = plane_sector.draw_above_b ^ (side_b == LineSide::BELLOW);
+
+            if plane_sector.sweep < ANGLE_180DEG {
+                correct_a_side && correct_b_side
+            } else if plane_sector.sweep < ANGLE_360DEG {
+                correct_a_side || correct_b_side
+            } else {
+                true
+            }
+        })
+    }
+}
+
+/// Iterator that returns the squared distance to the center for all points in the bounding box.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct DistanceIterator {
```

Comment:
We should merge this with `circle::DistanceIter`, by replacing `circle::DistanceIter` with:
(I didn't test the code, but it hope it's working enough to describe what I mean)
```rust
#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
pub(in crate::primitives) struct DistanceIterator<I> {
    center: Point,
    points: I,
}

impl<I> DistanceIterator<I>
where
    I: Iterator<Item = Point>
{
    pub(in crate::primitives) fn new(center: Point, points: I) -> Self {
        Self {
            center,
            points,
        }
    }
}

impl<I> Iterator for DistanceIterator<I>
where
    I: Iterator<Item = Point>
{
    type Item = (Point, u32);

    fn next(&mut self) -> Option<Self::Item> {
        self.points.next().map(|p| {
            let delta = self.center - p * 2;
            let distance = delta.x.pow(2) as u32 + delta.y.pow(2) as u32;

            (p, distance)
        })
    }
}
```
Which could then be used for `Circle`, `Arc` and `Sector`.